runtime/runtime.go

```diff
@@ -214,11 +218,21 @@ func (c ContainerData) Environment(ctx context.Context) map[string]string {
 // IP returns an IP adress associated with an container "label"
 func (c ContainerData) IP(label string) string {
 	if c.serviceCtx != nil {
-		for _, containerDef := range c.serviceCtx.ContainerDefinitions {
-			if ipv4, err := narwhal.IPv4Address(context.Background(), narwhal.DockerClient(), containerDef.Label); err == nil {
-				return ipv4.String()
+		var containerDef narwhal.ContainerDefinition
+		for _, c := range c.serviceCtx.ContainerDefinitions {
+			log.Debugf("Container Def label: %s; searching: %s", c.Label, label)
+			if label == c.Label {
+				containerDef = c
+				break
 			}
 		}
+		container, err := c.serviceCtx.FindContainer(context.Background(), &containerDef)
+		if err != nil {
+			return ""
+		}
+		if ipv4, err := narwhal.IPv4Address(context.Background(), c.serviceCtx.DockerClient, container.Id); err == nil {
```

Comment:
nit: This seems more like the `Context` isn't being plumbed, so let's make that explicit:

```suggestion
		container, err := c.serviceCtx.FindContainer(context.TODO(), &containerDef)
		if err != nil {
			return ""
		}
		if ipv4, err := narwhal.IPv4Address(context.TODO(), c.serviceCtx.DockerClient, container.Id); err == nil {
```

Reply:
FYI, I can't think of any way to plumb in a `Context` here, as this function is called when interpolating `{{ .Containers.IP "xxx" }}`. If I add this, it'd stop working. Or am I missing something here?

Reply:
You are correct, but I'm saying that `context.TODO` is the correct function to use when we really ought to be plumbing a `Context`, but can't think of a good way to do it right away.